sagemaker-debugger/model_specific_realtime_analysis/autoencoder_mnist/autoencoder_mnist.ipynb

```diff
@@ -0,0 +1,418 @@
+    "<img src=\"images/tsne.gif\" alt=\"drawing\" width=\"600\"/> "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training MXNet autoencoder model in Amazon SageMaker with debugger \n",
+    "Before starting the SageMaker training job, we need to install some libraries. We will use `smdebug` library to read, filter and analyze raw tensors that are stored in Amazon S3. We install `seaborn` library that will be used later on to plot t-Distributed Stochastic Neighbor Embedding (t-SNE) of the latent variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install smdebug\n",
+    "! pip install -U seaborn"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First we define the MXNet estimator and the debugger hook configuration. The model training is implemented in the entry point script `autoencoder_mnist.py`. We will obtain tensors every 10th iteration and store them in the SageMaker default bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "from sagemaker.mxnet import MXNet\n",
+    "from sagemaker.debugger import DebuggerHookConfig, CollectionConfig\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "BUCKET_NAME = sagemaker_session.default_bucket()\n",
+    "LOCATION_IN_BUCKET = 'smdebug-autoencoder-example'\n",
+    "\n",
+    "s3_bucket_for_tensors = 's3://{BUCKET_NAME}/{LOCATION_IN_BUCKET}'.format(BUCKET_NAME=BUCKET_NAME, LOCATION_IN_BUCKET=LOCATION_IN_BUCKET)\n",
+    "estimator = MXNet(role=sagemaker.get_execution_role(),\n",
+    "                  base_job_name='mxnet',\n",
+    "                  train_instance_count=1,\n",
+    "                  train_instance_type='ml.m5.xlarge',\n",
+    "                  train_volume_size=400,\n",
+    "                  source_dir='src',\n",
+    "                  entry_point='autoencoder_mnist.py',\n",
+    "                  framework_version='1.6.0',\n",
+    "                  py_version='py3',\n",
+    "                  debugger_hook_config = DebuggerHookConfig(\n",
+    "                      s3_output_path=s3_bucket_for_tensors,  \n",
+    "                      collection_configs=[\n",
+    "                        CollectionConfig(\n",
+    "                            name=\"all\",\n",
+    "                            parameters={\n",
+    "                                \"include_regex\": \".*convolutionalautoencoder0_hybridsequential0_dense0_output_0|.*convolutionalautoencoder0_input_1|.*loss\",\n",
```

Comment:
Hi, how do you know the names of this layer? what is the function you used to get that? 

And also, i can not get how Sagemaker debug can hook in the model? If i own the training script should i do something with the sagemaker debug api to register my model? I can not see the code that link the model that i create in the script to the sagemaker debug APi.
Many thanks.